### ISSUE TITLE
NVDA does not read labels on web components that are in a shadow root

### ISSUE THREAD

**Issue by jesmrtn1** [nvaccess/nvda#13890]
### Is your feature request related to a problem? Please describe.

We are currently adding accessibility to a tooltip but we are having issues getting NVDA to read the text in tooltip when we are on the button. Voice over reads it just fine. Below there is an example om how our button and toast looks like. It’s strange that it works with voice over but not NVDA.

The problem lies in the ShadowDOM after all since its boundary breaks the connection between the button and the tooltip. We've tried using aria-live and the combination of aria-describedBy and a referencing id. Since the toast and the button are rendered in their own components with their own shadow root, they are not aware of each other, which is why NVDA fails to make the connection. It would be nice to find a solution for this known-problem.

<kompis-tooltip theme="${theme}" text="This is a tooltip">
	<kompis-button label="Hover over to show tooltip" theme="${theme}" type="primary">
         </kompis-button>
</kompis-tooltip>

**Comment by seanbudd** [comment]
Welcome @jesmrtn1 ,

We are unable to process this issue as it stands.

- This appears to be a bug report not a feature request.
Please fill out the [bug report template](https://github.com/nvaccess/nvda/issues/new?assignees=&labels=&template=bug_report.md)
- To be able to reproduce this issue, we need a code sample. Provide a minimal HTML sample to reproduce this using [codepen](https://codepen.io/).

Kind Regards,
NV Access Software Developers

**Comment by Brian1Gaff** [comment]
Its interesting because something very similar is nowpart of the released 
new version of Dropbox, where their nice little alerts are no longer seen by 
nvda, nor is the new context menu most of the time it seems.
  This isDropbox, as nvda has not changed, but was so similar as to make me 
wonder if the two might be related in some way. Also witness the Team Viewer 
lack of access when I understand Jaws can still see the  objects.
 Brian

-- 
***@***.***
Sent via blueyonder.(Virgin media)
Please address personal E-mail to:-
***@***.***, putting 'Brian Gaff'
in the display name field.
----- Original Message -----
